language/vm/src/errors.rs

```diff
@@ -62,12 +62,11 @@ impl VMError {
                 VMStatus::Error(StatusCode::ABORTED)
             }
 
-            // TODO Errors for OUT_OF_GAS do not always have index set
             (major_status, sub_status, location)
                 if major_status.status_type() == StatusType::Execution =>
             {
                 debug_assert!(
-                    offsets.len() == 1,
+                    offsets.len() == 1 || major_status == StatusCode::OUT_OF_GAS,
```

Comment:
I would comment for this case.

And you likely also need to fix the asserts in `VMStatus::status_code`

Overall, this approach does bother me a bit as it breaks the bucketing scheme and ruins any of the asserts for this code, i.e. if we break the logging for it in the interpreter it becomes a lot harder to detect

Reply:
Let me look into the option of splitting into 2 different OUT_OF_GAS codes....

Reply:
I think somewhat conceptually this makes sense to me. But might be a breaking change (which could be bad)
There is:
- Out of gas from the Move semantics
- Out of gas from the practical Diem side of things
Move out of gas implies Diem out of gas, but not the other way around, which is the issue this assert caught (in a way)

Reply:
cc @tzakian

Reply:
I looked into splitting out a new OUT_OF_GAS code that isn't an execution failure, but such a status code wouldn't fit in any of the other existing error buckets. I'm not sure it makes sense to add a new bucket for this. I'm going to think about this some more....